x/delegation/keeper/logic_f1distribution.go

```diff
@@ -289,15 +290,15 @@ func (k Keeper) f1GetOutstandingRewards(ctx sdk.Context, stakerAddress string, d
 	// delegation amount for the period.
 	// To incorporate slashing one needs to iterate all slashes and calculate the reward for every period
 	// separately and then sum it.
-	reward := math.LegacyNewDec(0)
+	reward := sdk.NewDecCoins()
 	latestBalance := math.LegacyNewDec(int64(delegator.InitialAmount))
 	k.f1IterateConstantDelegationPeriods(ctx, stakerAddress, delegatorAddress, delegator.KIndex, endIndex,
 		func(startIndex uint64, endIndex uint64, delegation math.LegacyDec) {
 			difference := k.f1GetEntryDifference(ctx, stakerAddress, startIndex, endIndex)
 			// Multiply with delegation for period
-			periodReward := difference.Mul(delegation)
+			periodReward := difference.MulDec(delegation)
```

Comment:
For me multiplying coins with zero panics with

```
multiplying by zero is an invalid operation on coins

  Full Stack Trace
    github.com/cosmos/cosmos-sdk/types.Coins.MulInt(...)
```

Because of this I had to do an extra zero check. Can this happen here if a delegator has exactly zero ukyve delegated?

Reply:
Thanks, I dont think so but I also the method can not know this. I will add it